src/decaf.cpp

```diff
@@ -318,4 +344,15 @@ namespace decaf {
 
     return expr;
   }
+
+  z3::expr normalize_to_uint(const z3::expr& expr) {
```

Comment:
This method is exactly the same as `normalize_to_int`? Note that LLVM doesn't have a difference between signed and unsigned integers, they're told apart by using different instructions instead.

Reply:
There is a slight difference. Since I pass an unsigned int to `ctx.bv_val` it internally calls `z3_make_unsigned`

See [z3 ref](https://z3prover.github.io/api/html/classz3_1_1context.html#ae91d68c90fd91bc89e30c2bbbff53f22)

Reply:
There's no difference in the end result though? Its still either getting back a 1bit integer with value 1 or 0. Making it from a signed int or not doesn't change that.

If you're still concerned then I'd just change the original method to use unsigned values.